RISE-Dependable-Transport-Systems/WayWiseR
# Feature/perception

> * **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

> * **What is the current behavior?** (You can also link to an open issue here)

Does not have object detection functionality.

> * **What is the new behavior (if this is a feature change)?**

Provides object detection capabilities using YOLOv8.

> * **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

>* **Other information**:

To test:
1. ensure that the waywiser_perception package requirements are met, according to the instructions in [README.md](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/tree/feature/perception/waywiser_perception#requirements).
2. setup camera:
     ### simulated camera in the gazebo:
    - Open a new terminal and run: `ros2 launch waywiser_gazebo gazebo.launch.py world:=./src/WayWiseR/waywiser_gazebo/worlds/bounded_world.sdf`
    - Open a new terminal and run: `ros2 launch waywiser_gazebo spawn.launch.py model_sdf_paths:='["./src/WayWiseR/waywiser_description/sdf/actor_stand/model.sdf", "./src/WayWiseR/waywiser_description/sdf/drone/model.sdf"]' `
    ### realsense d435i camera on jetson:
    - set 'enable_camera' to true in [rover.yaml](https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/b87feb96883e69df4098843bdf3388ef5672eb0a/waywiser_hwbringup/config/rover.yaml#L25).
    - Open a new terminal and run: `ros2 launch waywiser_hwbringup rover.launch.py`

3. Open a new terminal and run: `ros2 launch waywiser_perception yolov8.launch.py use_sim_time:=false`
4. Open a new terminal and run: `ros2 launch waywiser_rviz2 rviz.launch.py use_sim_time:=false`. Expand the left panel and select 'Object detection', 'Color pointcloud', and '3D bounding box markers'.
